### PR TITLE
Extend docs on how to support PyMC

### PR DESCRIPTION
**Pull request by OriolAbril**
This extends README, docs home page and most significantly reconverts the history page to an about page, with a new section on supporting PyMC. It tries to highlight the different alternatives to support pymc, especially the brand new tidelift option. 

There are still a couple points to polish though.
* For tidelift `learn more`/`request a demo` button, is a button or a badge better? 
* Is it worth it to have the long description on tidelift subscription, should we modify it to make it more presonalized to pymc? For now I have just copy pasted the template provided by tidelift

Note: see the proposed [docs homepage](https://oriolabril.github.io/pymc3/index.html) and [about page](https://oriolabril.github.io/pymc3/about.html#support) in my fork's gh-pages

### CODE REVIEW

**Comment by AlexAndorra** [comment]
I like the idea, thanks Oriol!
- I don't see the difference with the current homepage though 🤔 What am I missing?
- Tidelift "learn more/demo": I prefer the buttons -- more prominent, you want to click it 😜 

> Is it worth it to have the long description on tidelift subscription, should we modify it to make it more presonalized to pymc? 

I think the long description is useful, and it's fine having the classic Tidelift template. The shorter description though, the one above the buttons, could gain from being tailored to PyMC (I'm guessing most people will read this one).

Also, I'm wondering whether it'd be better to have a dedicated "Support" page that we'd add to the navbar. There we'd put both the ways private users can support PyMC, and also the Enterprise subscriptions. Right now, the about page is a mix of "Here's how to use PyMC" and "Here's how to support PyMC", so it could heart readbility. What do you think?

**Comment by OriolAbril** [comment]
> I don't see the difference with the current homepage though thinking What am I missing?

The changes in the home page are quite small, I have added a link (which won't work in my fork) in the text of the support section and I have made all sponsor logos to have the same size

> I think the long description is useful, and it's fine having the classic Tidelift template. The shorter description though, the one above the buttons, could gain from being tailored to PyMC (I'm guessing most people will read this one).

Feel free to propose changes to the small description. What feels most weird to me though is in the long description, having _we_ refer to tidelift and _the creators behind the software you use_ being pymc devs

> Also, I'm wondering whether it'd be better to have a dedicated "Support" page that we'd add to the navbar.

I was initially planning on this, but it does not look like there is any extra space in the navbar :confused:

**Comment by AlexAndorra** [comment]
> The changes in the home page are quite small, I have added a link (which won't work in my fork) in the text of the support section and I have made all sponsor logos to have the same size

Aaaah, I see it now! Looks good.

> Feel free to propose changes to the small description. What feels most weird to me though is in the long description, having we refer to tidelift and the creators behind the software you use being pymc devs

How is that?

- **Small description**: 

PyMC is now available as part of the Tidelift Subscription! 

Tidelift is working with us, the maintainers of PyMC, to deliver commercial support and maintenance for the open source dependencies companies use to build their applications. By subscribing to Tidelift, you save time, reduce risk and improve code health, while contributing financially to PyMC -- making it even more robust, reliable and, let's face it, amazing!

- **Big description**:

**Enterprise-ready open source software—managed for you**

The Tidelift Subscription is a managed open source subscription for application dependencies covering millions of open source projects across JavaScript, Python, Java, PHP, Ruby, .NET, and more. And now, your favorite probabilistic programming language is included in the Tidelift subscription ;)

Your subscription includes:

- Security updates: Tidelift’s security response team coordinates patches for new breaking security vulnerabilities and alerts immediately through a private channel, so your software supply chain is always secure.

- Licensing verification and indemnification: Tidelift verifies license information to enable easy policy enforcement and adds intellectual property indemnification to cover creators and users in case something goes wrong. You always have a 100% up-to-date bill of materials for your dependencies to share with your legal team, customers or partners.

- Maintenance and code improvement: Tidelift ensures the software you rely on keeps working as long as you need it to work. Your managed dependencies are actively maintained and we recruit additional maintainers where required.

- Package selection and version guidance: Tidelift team helps you choose the best open source packages from the start -- and then guides you through updates to stay on the best releases as new issues arise.

- Roadmap input: Take a seat at the table with the creators behind the software you use. PyMC developers and other Tidelift’s participating maintainers earn more income as our software is used by more subscribers, so we're interested in knowing what you need.

- Tooling and cloud integration: Tidelift works with GitHub, GitLab, BitBucket, and more. They support every cloud platform (and other deployment targets too).

The end result? All of the capabilities you expect from commercial-grade software, for the full breadth of open source you use. That means less time grappling with esoteric open source trivia, and more time building your own applications—and your business.

- Also, for the "Support PyMC" section of the about page, I was thinking of modifying it like:
PyMC is a non-profit project under NumFOCUS umbrella. If you value PyMC and want to support its sustained and thorough development, consider donating to the project.

Hope this helps, and tell me if anything is unclear 😉  If easier for you, I can probably include these proposals as "suggested changes" in a review of this PR?

**Review by twiecki — APPROVED** [review]

**Comment by AlexAndorra** [comment]
Ow, and about that:
> I was initially planning on this, but it does not look like there is any extra space in the navbar 😕

There are themes where it's possible to have dropdowns in navbar items. Do you know if it's possible with this one?
Here for instance, we would have the "About PyMC" tab, and when clicking it a dropdown is triggered with two choices, "History & Usage" and "Support & Enterprise Plan". Do you see what I mean?

**Comment by OriolAbril** [comment]
Thanks! I'll start adding changes

> Also, for the "Support PyMC" section of the about page, I was thinking of modifying it like:
> PyMC is a non-profit project under NumFOCUS umbrella. If you value PyMC and want to support its sustained and thorough development, consider donating to the project.

Here I'm not sure where would the links to support page and to donation page in numfocus go :confused: 

Dropdown sounds good even if a little hidden, I don't know how to do this though, maybe @ColCarroll has some pointers?

File: docs/source/about.rst
Comment (AlexAndorra) [inline docs/source/about.rst]:
```suggestion
PHP, Ruby, .NET, and more. And now, your favorite probabilistic programming language is included in the Tidelift subscription!
```

**Review by AlexAndorra — COMMENTED** [review]
Nice, the changes look good, thanks @OriolAbril ! Just spotted a typo below.

> Here I'm not sure where would the links to support page and to donation page in numfocus go 😕

What you ended up writing is perfect 👌 

> Dropdown sounds good even if a little hidden, I don't know how to do this though

I did this [for my website](https://www.pollsposition.com/) (hover over the "A propos" section in the navbar):
```html
<li class="nav-item dropdown d-md-none d-lg-block">
    <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button"
          data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"
    >
        A propos
    </a>
    <div class="dropdown-menu" aria-labelledby="navbarDropdown">
        <a class="dropdown-item" href="/home/about">
            Qui sommes-nous ?
        </a>
        <a class="dropdown-item" href="/home/all_articles">
            Tous nos articles
        </a>
        <a class="dropdown-item" href="/home/friends">
            Amis du site
        </a>
    </div>
</li>
```
It's using the [Bootstrap](https://getbootstrap.com/) framework. Is it useful in any way?

**Comment by OriolAbril** [comment]
@AlexAndorra can you open an issue about modifying the nav bar? The links are currently defined in [`conf.py`](https://github.com/pymc-devs/pymc3/blob/master/docs/source/conf.py#L152) and I don't really know how to combine that with the html code above. 

Or if it's easier hijack the PR to see if you can modify it yourself

**Comment by AlexAndorra** [comment]
Ok, let's do that in another PR then. I'll merge this one once we figure out what the heck is going on with our Travis dev env. Thanks again!

**Review by AlexAndorra — APPROVED** [review]

**Comment by OriolAbril** [comment]
Thanks for merging master into it, everything looks good

**Comment by AlexAndorra** [comment]
De nada 😉 Let's see which color the tests turn out to be though...

**Comment by AlexAndorra** [comment]
¡Todo está verde! 💚 --> merging!

**Pull request merged**
